src/mito2/src/region.rs

```diff
@@ -297,19 +298,30 @@ impl MitoRegion {
     }
 
     /// Sets the staging state.
+    ///
     /// You should call this method in the worker loop.
     /// Transitions from Writable to Staging state.
-    pub(crate) fn set_staging(&self) -> Result<()> {
+    /// Cleans any existing staging manifests before entering staging mode.
+    pub(crate) async fn set_staging<'a, 'b>(
+        &'a self,
+        manager: RwLockWriteGuard<'b, RegionManifestManager>,
+    ) -> Result<()>
+    where
+        'a: 'b,
```

Comment:
non-blocking: maybe we can pass `&mut RegionManifestManager` from the guard.
```suggestion
    pub(crate) async fn set_staging(
        &self,
        manager: &mut RegionManifestManager,
    ) -> Result<()>
```